lib/perl/Genome/Sys/LockProxy.pm

```diff
@@ -0,0 +1,106 @@
+package Genome::Sys::LockProxy;
+
+use strict;
+use warnings;
+
+use Genome::Sys::Lock qw();
+use Params::Validate qw(validate_with);
+use Scalar::Util qw(blessed);
+
+=item new()
+
+Keyword Arguments:
+
+=over
+
+=item resource
+
+C<resource> is a unique identifier for some resource.
+
+=item scope
+
+C<scope> is the scope to which the lock is bound.  See C<Genome::Sys::Lock::scopes()> for valid scopes.
+
+=cut
+
+sub new {
+    my $class = shift;
+    my $spec = Genome::Sys::Lock::LOCK_RESOURCE_PARAMS_SPEC();
+    $spec->{resource} = delete $spec->{resource_lock};
+    my %params = validate_with(
+        params => \@_,
+        spec => $spec,
+    );
+    return bless \%params, $class;
```

Comment:
It seems like it would be silly if someone called `new()` on an object.  Is that a thing?  I am checking that it's not a ref so instead of,

```
Attempt to bless into a reference at /gscuser/nnutter/genome/lock-pair/lib/perl/Genome/Sys/LockProxy.pm line 32.
```

you'll get,

```
new() should be called as a class method at -e line 1
```

but maybe I should just make it bless into the class of the object?